src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/VariantFmsDTOValidator.java

```diff
@@ -0,0 +1,325 @@
+			throw new ObjectValidationException(dto, variantResponse.errorMessagesString());
+		}
+		
+		variant = variantDAO.persist(variant);
+		if (variant != null) {
+			idsAdded.add(variant.getId());
+		}
+
+		return variant;
+	}
+	
+	@Transactional
+	public void validateCuratedVariantGenomicLocationAssociation(VariantFmsDTO dto, List<Long> idsAdded, Variant variant) throws ValidationException {
+
+		CuratedVariantGenomicLocationAssociation association = new CuratedVariantGenomicLocationAssociation();	
+		ObjectResponse<CuratedVariantGenomicLocationAssociation> cvglaResponse = new ObjectResponse<CuratedVariantGenomicLocationAssociation>();
+		AssemblyComponent chromosome = null;
+		
+		if (dto.getStart() == null) {
+			cvglaResponse.addErrorMessage("start", ValidationConstants.REQUIRED_MESSAGE);
+		} 
+		
+		if (dto.getEnd() == null) {
+			cvglaResponse.addErrorMessage("end", ValidationConstants.REQUIRED_MESSAGE);
+		}
+		
+		if (StringUtils.isBlank(dto.getSequenceOfReferenceAccessionNumber())) {
+			cvglaResponse.addErrorMessage("sequenceOfReferenceAccessionNumber", ValidationConstants.REQUIRED_MESSAGE);
+		} else {
+			ChromosomeAccessionEnum cae = ChromosomeAccessionEnum.getChromosomeAccessionEnum(dto.getSequenceOfReferenceAccessionNumber());
+			if (cae != null) {
+				Map<String, Object> params = new HashMap<>();
+				params.put("name", cae.chromosomeName);
+				params.put("genomeAssembly.modEntityId", cae.assemblyIdentifier);
+				SearchResponse<AssemblyComponent> acResponse = assemblyComponentDAO.findByParams(params);
+				if (acResponse != null) {
+					chromosome = acResponse.getSingleResult();
+				}
+			}
+			if (chromosome == null) {
+				cvglaResponse.addErrorMessage("sequenceOfReferenceAccessionNumber", ValidationConstants.INVALID_MESSAGE + " (" +
+						dto.getSequenceOfReferenceAccessionNumber() + ")");
+			}
+		}
+		
+		String hgvs = HgvsIdentifierHelper.getHgvsIdentifier(dto);
+		
+		if (variant != null && StringUtils.isNotBlank(hgvs) && !cvglaResponse.hasErrors() && CollectionUtils.isNotEmpty(variant.getCuratedVariantGenomicLocations())) {
+			for (CuratedVariantGenomicLocationAssociation existingLocationAssociation : variant.getCuratedVariantGenomicLocations()) {
+				if (Objects.equals(hgvs, existingLocationAssociation.getHgvs())) {
+					association = curatedVariantGenomicLocationAssociationDAO.find(existingLocationAssociation.getId());
+					break;
+				}
+			}
+		}
+		
+		association.setHgvs(hgvs);
+		association.setVariantAssociationSubject(variant);
+		association.setVariantGenomicLocationAssociationObject(chromosome);
+		association.setStart(dto.getStart());
+		association.setEnd(dto.getEnd());
+		association.setRelation(vocabularyTermService.getTermInVocabulary(VocabularyConstants.LOCATION_ASSOCIATION_RELATION_VOCABULARY, "located_on").getEntity());
+		if (StringUtils.isNotBlank(dto.getGenomicReferenceSequence())) {
+			association.setReferenceSequence(dto.getGenomicReferenceSequence());
+		}		
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineJavaCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line must not have trailing whitespace
<!-- __reviewdog__:ChA2MTM3NGYxNjUxNzVhMjg4EgpjaGVja3N0eWxl -->